derive_stringset/src/lib.rs

```diff
@@ -70,3 +79,41 @@ pub fn derive(input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+fn derive_enum(input: &syn::DeriveInput) -> TokenStream {
+    let variants = if let syn::Data::Enum(syn::DataEnum { ref variants, .. }) = input.data {
```

Comment:
Why not take in `syn::Data::Enum` as the input args instead then? Would remove the necessity of this check entirely.